test/src/main/java/org/apache/accumulo/test/ComprehensiveTableOperationsIT.java

```diff
@@ -0,0 +1,957 @@
+    ops.create(userTable);
+    // system and user tables
+    Set<String> allTables = ops.list();
+    for (var table : allTables) {
+      // VersioningIterator is a default iterator
+      assertTrue(ops.testClassLoad(table, VersioningIterator.class.getName(),
+          SortedKeyValueIterator.class.getName()));
+      assertFalse(ops.testClassLoad(table, "foo", SortedKeyValueIterator.class.getName()));
+    }
+  }
+
+  @Test
+  public void test_setSamplerConfiguration_getSamplerConfiguration_clearSamplerConfiguration()
+      throws Exception {
+    // these are mostly untested. Test basic functionality here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+    // system and user tables
+    Set<String> allTables = ops.list();
+    for (var table : allTables) {
+      ops.setSamplerConfiguration(table, SampleIT.SC1);
+      assertEquals(SampleIT.SC1, ops.getSamplerConfiguration(table));
+      ops.clearSamplerConfiguration(table);
+      assertNull(ops.getSamplerConfiguration(table));
+    }
+  }
+
+  @Test
+  public void test_summaries_addSummarizers_removeSummarizers_listSummarizers() throws Exception {
+    // these are all tested for user tables in SummaryIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(SummaryIT.class.getName()));
+    // these are not tested for system tables. Test basic functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      SummarizerConfiguration sc = SummarizerConfiguration.builder(BasicSummarizer.class).build();
+
+      assertThrows(AccumuloSecurityException.class,
+          () -> ops.summaries(sysTable.tableName()).retrieve());
+      ops.addSummarizers(sysTable.tableName(), sc);
+      assertEquals(List.of(sc), ops.listSummarizers(sysTable.tableName()));
+      ops.removeSummarizers(sysTable.tableName(), sc1 -> true);
+      assertTrue(ops.listSummarizers(sysTable.tableName()).isEmpty());
+    }
+  }
+
+  @Test
+  public void test_getTimeType() throws Exception {
+    // getTimeType is tested elsewhere but not for all system tables. Simple enough to test all
+    // tables here
+    userTable = getUniqueNames(1)[0];
+    ops.create(userTable);
+
+    assertEquals(TimeType.MILLIS, ops.getTimeType(userTable));
+    for (var sysTable : AccumuloTable.values()) {
+      assertEquals(TimeType.LOGICAL, ops.getTimeType(sysTable.tableName()));
+    }
+  }
+
+  @Test
+  public void test_setTabletAvailability_getTabletInformation() throws Exception {
+    // these are tested for user tables in TabletAvailabilityIT. Ensure test exists
+    assertDoesNotThrow(() -> Class.forName(TabletAvailabilityIT.class.getName()));
+    // these are not tested for system tables. Test basic functionality here
+    for (var sysTable : AccumuloTable.values()) {
+      // should not be able to unhost any system table
+      assertThrows(IllegalArgumentException.class, () -> ops
+          .setTabletAvailability(sysTable.tableName(), new Range(), TabletAvailability.UNHOSTED));
```

Comment:
Should maybe be an AccumuloException to keep consistency with the other table operations methods that throw an AccumuloException when performed on an unacceptable table?

Reply:
Opened https://github.com/apache/accumulo/pull/5510

A lot of the table operations don't have javadoc for what the thrown exceptions mean, so maybe this can be it's own ticket

Reply:
Functionality fixed in #5510, assertion fixed in 937e4f8c595b128e94a32ab0e2b687a0ff8cfe31